env_store.go

```diff
@@ -98,56 +81,52 @@ func (s *EnvStore) GetSecretReferences() map[string][]string {
 // It then asynchronously loads secrets using each provider and it's corresponding paths.
 // The secrets from each provider are then placed into a single slice.
 func (s *EnvStore) LoadProviderSecrets(ctx context.Context, providerPaths map[string][]string) ([]provider.Secret, error) {
-	// At most, we will have one error per provider
-	errCh := make(chan error, len(supportedProviders))
 	var providerSecrets []provider.Secret
-
 	// Workaround for openBao
 	// Remove once openBao uses BAO_ADDR in their client, instead of VAULT_ADDR
-	vaultPaths, ok := providerPaths[vault.ProviderName]
-	if ok {
-		var err error
-		providerSecrets, err = s.workaroundForBao(ctx, vaultPaths)
+	if _, ok := providerPaths["vault"]; ok {
+		vaultSecrets, err := s.workaroundForBao(ctx, providerPaths["vault"])
 		if err != nil {
-			return nil, fmt.Errorf("failed to workaround for bao: %w", err)
+			return nil, err
 		}
 
-		// Remove the vault paths since they have been processed
-		delete(providerPaths, vault.ProviderName)
+		providerSecrets = append(providerSecrets, vaultSecrets...)
+		delete(providerPaths, "vault")
```

Comment:
if you need to use the specific provider type, id suggest to keep using the exposed consts (in this case, `vault.ProviderName`, or `vault.ProviderType` for usage clarity